### ISSUE TITLE
Set compare build id

### ISSUE THREAD

**Issue by Joannazhx** [rfennell/AzurePipelines#838]
Hi, Richard. 
In our release pipeline, the change may need to be rolled out for enterprise clients. In this case, I wish the work items could be getting from a specific build to the current one instead of the last successful one.

**Is your feature request related to a problem? Please describe.**
Considering work items get from Azure API get work items between builds 
```
 GET https://dev.azure.com/{organization}/{project}/_apis/build/workitems?fromBuildId={fromBuildId}&toBuildId={toBuildId}&api-version=6.0-preview.2
```
In log file
```
Getting the current build details
##[debug]System.StageName=__default
Getting items associated the builds since the last successful build to the stage '__default'
Comparing 132961 against 132961
Ignore compare against self
Comparing 132960 against 132961
Considering build
```

When I set check stage == true, it will be compared with the last successful build
But the last successful build may not run in the same branch as the current one.

**Describe the solution you'd like**
Could I set {fromBuildId} in the build pipeline?

**Additional context**
Also, I noticed the advanced Name of the stage (override) -- System.StageName
In multiple-stage pipeline could I set both {fromBuildId} and {System.StageName} ?
 
Thanks for your help,
Joanna.

**Comment by rfennell** [comment]
I can't see a technical reason to not add this feature. My only concern would be the operational one if it bit a bit of a manual overhead to keep updating the new `fromBuildId` parameter.

**Comment by nhojt** [comment]
+1 for this feature as a bit of safety net. A couple of times we've had dev's switch branches and the release notes go haywire as a result. It would be useful after that happens to be able specify what build to compare against (otherwise continue to default to current behaviour).

**Comment by rfennell** [comment]
The people have spoken, I will push this feature up the backlog

**Comment by rfennell** [comment]
I have implemented an override for both multistage YAML builds and UI based release. The new parameter (on the Advanced menu) accepts an build or release ID e.g:

```
overrideBuildReleaseId  = 1234
```

For releases this value is the release ID of the 'last good release' to compare against, for multi-stage YAML it is the build ID.

Note that 
- If this value is not specified, the default, then the good build/release logic is used
- If this value cannot be parsed to an int as entered, then the task will exit with an error
- If this specified release/build is not found then the task will exit with an error
- No check is made as to the success for failure if the specified build/release - it is assumed the administrator know what they are doing.

The new version is currently in test but will be released as 3.21.x
